types/react/index.d.ts

```diff
@@ -49,7 +52,7 @@ declare namespace React {
     }
 
     interface ReactElement<P> {
-        type: string | ComponentClass<P> | SFC<P>;
+        type: string | keyof ReactHTML | keyof ReactSVG | ComponentClass<P> | SFC<P>;
```

Comment:
this addition doesn't do anything since `string` is a supertype of `keyof ReactHTML` and `keyof ReactSVG`.